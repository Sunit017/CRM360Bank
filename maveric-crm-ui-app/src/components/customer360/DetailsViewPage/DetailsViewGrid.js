import { Grid, Box } from "@mui/material";
import styled from "styled-components";

const Label = styled.div`
  marginright: 5px;
  display: flex;
  font-weight: 700;
  color: #68077b;
  // border: 1px solid red;
`;

const DetailsViewGrid = ({ viewDetails, resData }) => {
  const listColumn = viewDetails?.children;

  return (
    <>
      <Box
        data-testid="DetailsViewGrid"
        sx={{
          width: "100%",
          // maxWidth: 300,
          position: "relative",
          overflow: "auto",
          maxHeight: "calc(100vh - 200px)",
          borderRadius: "none",
          "&::-webkit-scrollbar": {
            width: 4,
            height: 4,
          },
          "&::-webkit-scrollbar-track": {
            backgroundColor: "#ced4da",
          },
          "&::-webkit-scrollbar-thumb": {
            backgroundColor: "#68077b",
            borderRadius: 100,
          },
        }}
      >
        <Grid
          container
          spacing={{ xs: 2, md: 3, lg: 1 }}
          columns={{ xs: 4, sm: 8, md: 12 }}
          data-testid="test-2"
        >
          {resData && Object.keys(resData).length>0
            ? Object.keys(resData).map((itmKey, index) => {
                const labelItem = listColumn.filter(
                  (item) => item.id === itmKey
                )[0];
                // const label = labelItem?.label || camelCaseToTitleCase(itmKey);
                const label = labelItem?.label
                return itmKey !== "addressId" && itmKey !== "relationshipId" && label ? (
                  <Grid item xs={2} sm={4} md={4} key={itmKey}>
                    <Box>
                      <Label>{label} :</Label>
                      {resData && resData[itmKey].toString()}
                    </Box>
                  </Grid>
                ) : null;
              })
            : (listColumn || []).map((item, index) => {
                const { id, label } = item;
                return (
                  <Grid item xs={2} sm={4} md={4} key={id}>
                    <Box>
                      <Label>{label} :</Label>
                      {resData && resData[id]}
                    </Box>
                  </Grid>
                );
              })}
        </Grid>
      </Box>
    </>
  );
};
export default DetailsViewGrid;
