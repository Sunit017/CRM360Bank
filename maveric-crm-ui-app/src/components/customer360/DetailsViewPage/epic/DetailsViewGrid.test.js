import DetailsViewGrid from "../DetailsViewGrid.js";
import renderWithRedux from "../../../../testUtil";
import { render, screen, cleanup } from "@testing-library/react";

afterEach(cleanup);

const pageDetails = {
  id: "test1",
  label: "Test Data",
  children: [
    { id: "1", label: "Test ID field" },
    {
      id: "2",
      label: "Test First Name field",
    },
    {
      id: "3",
      label: "Test Last Name field",
    },
  ],
};

const response = {
  1: "Test value 1",
  2: "Test value 2",
};

test("Render details view grid container", () => {
  const container = renderWithRedux(<DetailsViewGrid />, {});
  expect(container).toMatchSnapshot();
});

test("Renders details view grid correctly", () => {
  render(<DetailsViewGrid viewDetails={pageDetails} values={response} />);
  const detailsViewGrid = screen.getByTestId("DetailsViewGrid");
  expect(detailsViewGrid).toBeInTheDocument();
});

test("Renders details view grid key-value pair correctly-1", () => {
  render(<DetailsViewGrid viewDetails={pageDetails} values={response} />);
  const detailsViewGrid = screen.getByTestId("test-2");
  expect(detailsViewGrid).toBeInTheDocument();
  expect(detailsViewGrid).toHaveTextContent("Test ID field");
});

test("Renders details view grid key-value pair correctly-2", () => {
  render(<DetailsViewGrid viewDetails={pageDetails} values={response} />);
  const detailsViewGrid = screen.getByTestId("test-2");
  expect(detailsViewGrid).toBeInTheDocument();
  expect(detailsViewGrid).toHaveTextContent("Test First Name field");
  // expect(detailsViewGrid).toHaveTextContent("Test value 2");
});

test("Renders details view grid key-value pair snapshot correctly", () => {
  render(<DetailsViewGrid viewDetails={pageDetails} values={response} />);
  const detailsViewGrid = screen.getByTestId("test-2");
  expect(detailsViewGrid).toMatchSnapshot();
});
