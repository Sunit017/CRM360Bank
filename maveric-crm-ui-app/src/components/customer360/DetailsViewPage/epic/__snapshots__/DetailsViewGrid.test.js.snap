// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render details view grid container 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1c17ife"
        data-testid="DetailsViewGrid"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-spacing-md-3 MuiGrid-spacing-lg-1 css-vobqoz-MuiGrid-root"
          data-testid="test-2"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1c17ife"
      data-testid="DetailsViewGrid"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-spacing-md-3 MuiGrid-spacing-lg-1 css-vobqoz-MuiGrid-root"
        data-testid="test-2"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`Renders details view grid key-value pair snapshot correctly 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-spacing-md-3 MuiGrid-spacing-lg-1 css-vobqoz-MuiGrid-root"
  data-testid="test-2"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="sc-bczRLJ fHPAUt"
      >
        Test ID field
         :
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="sc-bczRLJ fHPAUt"
      >
        Test First Name field
         :
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-8v0fa4-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="sc-bczRLJ fHPAUt"
      >
        Test Last Name field
         :
      </div>
    </div>
  </div>
</div>
`;
